bsb/morphologies/__init__.py

```diff
@@ -1419,33 +1420,54 @@ def _line_dists(self, start, end):
             return np.linalg.norm(self.points - start, axis=1)
 
         vec = end - start
-        cross = np.cross(vec, start - self.points, axisb=0, axis=0)
-        return np.divide(np.linalg.norm(cross, axis = 0), np.linalg.norm(vec))
+        cross = np.cross(vec, start - self.points)
+        return np.divide(np.linalg.norm(cross, axis = 1), np.linalg.norm(vec))
 
 
     def simplify(self, epsilon, idx_start = 0, idx_end = -1):
-        if len(self.points) == 0:
-            return np.array([])
+        """
+        Apply Ramer–Douglas–Peucker algorithm to all points or a subset of points of the branch.
+        :param epsilon: Epsilon to be used in the algorithm.
+        :param idx_start = 0: Index of the first element of the subset of points to be reduced.
+        :param epsilon = -1: Index of the last element of the subset of points to be reduced.
+        """
+        if len(self.points) < 3:
+            return
         if idx_end == -1:
             idx_end = len(self.points) - 1
         if epsilon < 0:
             raise ValueError(f"Epsilon must be an int >= 0, actual epsilon: {epsilon}")
 
-        start, end = self.points[idx_start], self.points[idx_end]
-        dists = self._line_dists(start, end)
-
-        index = np.argmax(dists)
-        dmax = dists[index]
+        reduced = deque()
+        skipped = deque()
+        should_loop = True
         
-        if dmax > epsilon and idx_end - idx_start > 1:
-            result1 = self.simplify(epsilon, idx_start, index - 1)
-            result2 = self.simplify(epsilon, index, idx_end)
+        while len(skipped) > 0 or should_loop:
+            start = self.points[idx_start]
+            end = self.points[idx_end]
+            dists = self._line_dists(start, end)
+            index = np.argmax(dists)
+            dmax = dists[index]
+            
+            should_loop = False
+            if dmax > epsilon and idx_end - idx_start > 1:
+                skipped.append([index, idx_end])
+                idx_end = index - 1
+                reduced.append(idx_start)
+                reduced.append(idx_end)
+            elif len(skipped) > 0:
+                idx_pair = skipped.popleft()
+                idx_start = idx_pair[0]
+                idx_end = idx_pair[1]
```

Comment:
And this is where you SHOULD use tuple unpacking ;)

```suggestion
                idx_start, idx_end = skipped.popleft()
```

Reply:
Are you sure you have to append right and pop left? That's going to lead to `O(n)` stack operations instead of `O(1)`. Avoid it if you can. I'll think about how you could flip this around and let you know